src/shogun2-core/src/test/java/de/terrestris/shogun2/service/AbstractSecuredPersistentObjectServiceTest.java

```diff
@@ -0,0 +1,543 @@
+		assertEquals(existingPermissionCollection.getPermissions().size(), implToTest.getUserPermissions().get(user).getPermissions().size());
+
+		// mock
+		doReturn(newPermissionCollection).when(permissionCollectionService).saveOrUpdate(any(PermissionCollection.class));
+
+		// invoke method to test
+		crudService.addUserPermissions(implToTest, user, newPermission);
+
+		// be sure that the permission collection, BUT NOT the entity have been saved
+		verify(permissionCollectionService, times(1)).saveOrUpdate(any(PermissionCollection.class));
+		verify(dao, times(0)).saveOrUpdate(implToTest);
+
+		// assert that we have permission for exactly one user
+		assertEquals(1, implToTest.getUserPermissions().keySet().size());
+
+		// assert that we have set the correct number of permissions
+		assertEquals(existingPermissionCollection.getPermissions().size(), implToTest.getUserPermissions().get(user).getPermissions().size());
+	}
+
+	/**
+	 *
+	 */
+	@Test
+	public void removeUserPermissions_shouldDoNothingWhenPassedEntityIsNull() {
+
+		Permission permissions = Permission.ADMIN;
+		User user = new User("Dummy", "Dummy", "dummy");
+
+		crudService.removeUserPermissions(null, user , permissions);
+
+		// be sure that nothing happened
+		verify(permissionCollectionService, times(0)).saveOrUpdate(any(PermissionCollection.class));
+		verify(dao, times(0)).saveOrUpdate(any(getCrudService().getEntityClass()));
+
+		assertTrue(implToTest.getUserPermissions().keySet().isEmpty());
+	}
+
+	/**
+	 *
+	 */
+	@Test
+	public void removeUserPermissions_shouldDoNothingWhenNoPermissionsHaveBeenPassed() {
+
+		User user = new User("Dummy", "Dummy", "dummy");
+
+		crudService.removeUserPermissions(implToTest, user);
+
+		// be sure that nothing happened
+		verify(permissionCollectionService, times(0)).saveOrUpdate(any(PermissionCollection.class));
+		verify(dao, times(0)).saveOrUpdate(implToTest);
+
+		assertTrue(implToTest.getUserPermissions().keySet().isEmpty());
+	}
+
+	/**
+	 *
+	 */
+	@Test
+	public void removeUserPermissions_shouldDoNothingWhenNoPermissionsExist() {
+
+		final Permission writePermission = Permission.WRITE;
+
+		User user = new User("Dummy", "Dummy", "dummy");
+
+		crudService.removeUserPermissions(implToTest, user, writePermission);
+
+		// be sure that nothing happened
+		verify(permissionCollectionService, times(0)).saveOrUpdate(any(PermissionCollection.class));
+		verify(dao, times(0)).saveOrUpdate(implToTest);
+
+		assertEquals(0, implToTest.getUserPermissions().keySet().size());
+	}
+
+	/**
+	 *
+	 */
+	@Test
+	public void removeUserPermissions_shouldRemoveExistingPermission() {
+
+		final Permission readPermission = Permission.READ;
+		final Permission writePermission = Permission.WRITE;
+
+		PermissionCollection existingPermissionCollection = new PermissionCollection();
+
+		existingPermissionCollection.getPermissions().add(readPermission);
+		existingPermissionCollection.getPermissions().add(writePermission);
```

Comment:
With regard to the above comment: If you remove the above line, does this test still pass? And should it?